core-rust/state-manager/src/metrics.rs

```diff
@@ -211,6 +218,31 @@ impl CommittedTransactionsMetrics {
     }
 }
 
+impl VertexPrepareMetrics {
+    pub fn new(registry: &Registry) -> Self {
+        Self {
+            proposal_transactions_size: IntGauge::with_opts(opts(
```

Comment:
These feel a little weird to be gauges - I imagine you want to use a (basic) Summary instead, or a Histogram?

Reply:
I'd go for a single `wasted_proposal_space_size: Histogram` (I am assuming we already have other metrics for count and size distribution of committed transactions).

The gauge would give you a an exact value of a _very_ rare sample - I really believe that knowing an accurate-enough distribution of all samples is better.

Reply:
We do have for size and count of committed **transactions**. This captures proposals. I'm interested in both how big (or small) the proposals are, as well as the wasted bandwidth.

A Histogram indeed makes more sense.

Reply:
The point is that a Gauge metric isn't useful in this context, as it gives a snapshot of the last quantity when the metric was read, but really you want some kind of summary of things, for two reasons:
* In case multiple things happened between reads
* So you can sensibly aggregate distributions over time